bluepyopt/ephys/locations.py

```diff
@@ -392,6 +393,73 @@ def instantiate(self, sim=None, icell=None):
         return self.find_icomp(sim, branches)
 
 
+class NrnTrunkSomaDistanceCompLocation(NrnSecSomaDistanceCompLocation):
+    """Location at a distance from soma along a main direction.
+
+    We search for the section that is the furthest away from some along
+    a direction, and pick a location at a given distance from soma along
+    the path to that section.
+
+    If direction == 'radial', the largest radial direction is used.
+
+    This is most useful to follow the trunk of an apical dendrite
+    without knowing the apical point, but only that apical trunk goes along y.
+    """
+
+    def __init__(
+        self,
+        name,
+        soma_distance=None,
+        sec_index=None,
+        sec_name=None,
+        direction=None,
+        comment=""
+    ):
+        """Constructor
+
+        Args:
+            name (str): name of this object
+            soma_distance (float): distance from soma to this compartment
+            sec_index (int): index of the section  to consider
+            sec_name (str): name of Neuron sections (ex: 'apic')
+            direction (list of 3 elements): 3d vector representing direction
+        """
+        super(NrnTrunkSomaDistanceCompLocation, self).__init__(
+            name,
+            soma_distance=soma_distance,
+            sec_index=sec_index,
+            sec_name=sec_name,
+            comment=comment
+        )
+
+        if direction is None:
+            direction = [0.0, 1.0, 0.0]
```

Comment:
Could you mention this default value in the docstring with the arguments, this is not trivial.